src/track/beatgrid.cpp

```diff
@@ -369,3 +388,34 @@ void BeatGrid::setBpm(double dBpm) {
     locker.unlock();
     emit updated();
 }
+
+void BeatGrid::setSignature(mixxx::TimeSignature sig, double dSample) {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return;
+
+    m_grid.mutable_first_beat()->mutable_signature()->set_beats(sig.getBeats());
+    m_grid.mutable_first_beat()->mutable_signature()->set_note_value(sig.getNoteValue());
+}
+
+mixxx::TimeSignature BeatGrid::getSignature(double dSample) const {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return mixxx::TimeSignature(0,0);
+
+    // If no signature assume 4/4
+    if (!m_grid.first_beat().has_signature()) {
+        return mixxx::TimeSignature(4,4);
```

Comment:
I can imagine that we have tracks with no signature, like speech intermissions or such. 
So I would prefer to not adopt 4/4 here. Can we adopt this as default when creating the BeatGrid? 

Alternative we may use this state as default = 4/4 and introduce a special value for has_signature()

Reply:
Interesting question. I recognized that I have my own bias as House and Techno DJ.
I will like to have inputs from the rest of the team.

Reply:
@daschuer After thinking on that topic I consider that is better to have a 4/4 signature as the default one.
Right now time signature has to be informed manually. I mean that we have no analyzer giving that information. Giving a no signature as default will force to set the time signature for every imported track manually.
I think is better to have 4/4 as default value and allow the user to modify the ones that are not 4/4, assuming that this is going to be a minority.
A different topic is to have the possibility to mark a track as "non music". May be is worth to raise the point in Zulip for further discussion. Your thoughts?